frontend/src/components/reports/routine/ActivityReportByPanel.js

```diff
@@ -0,0 +1,148 @@
+import PanelSelectForm from "../../workplan/PanelSelectForm";
+
+const ActivityReportByPanel = () => {
+  const intl = useIntl();
+  const mounted = useRef(false);
+  const [panels, setPanels] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [notificationVisible, setNotificationVisible] = useState(false);
+  const [setValue, setSetValue] = useState("");
+  const [reportFormValues, setReportFormValues] = useState({
+    startDate: null,
+    endDate: null
+  });
+  const [panelList, setPanelList] = useState([]);
+
+  const encodeDate = (dateString) => {
+    if (typeof dateString === "string" && dateString.trim() !== "") {
+      return dateString.split("/").map(encodeURIComponent).join("%2F");
+    } else {
+      return "";
+    }
+  };
+
+  const handleChangeDatePicker = (datePicker, date) => {
+    const updatedDate = encodeDate(date);
+    setReportFormValues(prevState => ({
+      ...prevState,
+      [datePicker]: updatedDate
+    }));
+  };
+
+  const handleSubmit = () => {
+    setLoading(true);
+    const baseParams = "report=activityReportByPanel&type=indicator";
+    const baseUrl = `${config.serverBaseUrl}/ReportPrint`;
+    const url = `${baseUrl}?${baseParams}&lowerDateRangeupperDateRange=${reportFormValues.startDate}&upperDateRange=${reportFormValues.endDate}`;
+    window.open(url, '_blank');
+    setLoading(false);
+    setNotificationVisible(true);
+  };
+
+  const handleSetValue = (v) => {
+    if (mounted.current) {
+      setSetValue(v);
+    }
+  };
+
+  useEffect(() => {
+    mounted.current = true;
+    const fetchPanelList = async () => {
+      try {
+        const data = await getFromOpenElisServer("/rest/panels");
+        console.log("Panel list:", data); 
+        setPanelList(data);
+      } catch (error) {
+        throw new Error("Error fetching panel list:", error);
+      }
+    };
+
+    fetchPanelList();
+
+    return () => {
+      mounted.current = false;
+    };
+  }, []);
+
+  return (
+    <>
+      <FormLabel>
+        <Section>
+          <Section>
+            <h1>
+              <FormattedMessage id="Activity report By panel"/>
+            </h1>
+          </Section>
+        </Section>
+      </FormLabel>
+      {notificationVisible && <AlertDialog />}
+      {loading && <Loading />}
+      <Grid fullWidth={true}>
+        <Column lg={16}>
+          <Form>
+            <Grid fullWidth={true}>
+              <Column lg={10}>
+                <Section>
+                  <br />               
+                  <h5>
+                    <FormattedMessage id="select date Range" />
+                  </h5>
+                </Section>
+                <div className="inlineDiv">
+                  <CustomDatePicker
+                    id={"startDate"}
+                    labelText={intl.formatMessage({
+                      id: "eorder.date.start",
+                      defaultMessage: "Start Date",
+                    })}
+                    autofillDate={true}
```

Comment:
@sherrif10 youll realise all the three reports have alot of simillar componenets ie date inputs. 

Just have a single component and render the reports specific selector based on the report type

Reply:
Alright Thanks.What is your suggestion about this output. Is this how its supposed to be.

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/48015736/f73a2246-8296-4566-8e02-0d45aa077c92

Reply:
Basing on your response, i will have only one component. ActivityReport.